skr-api/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM rust:1.71.0-slim-bullseye as build
+RUN apt-get update && apt-get install -y libssl-dev pkg-config protobuf-compiler curl gnupg2 build-essential clang
+RUN curl -L https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add -
+RUN echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu focal main' > /etc/apt/sources.list.d/intel-sgx.list
+RUN apt-get update && apt-get install -y libtdx-attest-dev libtss2-dev
+COPY Cargo.* /build/
+COPY build.rs /build/
+COPY protos /build/protos
+COPY src /build/src
+WORKDIR /build
+RUN cargo b --release --bin skr-api
+
+FROM debian:bullseye-slim as libraries
+RUN apt-get update && apt-get install -y libtss2-tctildr0 libtss2-esys-3.0.2-0
+
+FROM gcr.io/distroless/cc-debian11:latest
```

Comment:
Is it possible to dump the static binary into a scratch base image?

Reply:
yes, if it's a musl build that should work.

Reply:
yup, it works (fixed in [this commit](https://github.com/confidential-containers/cloud-api-adaptor/pull/1237/commits/d02a93550b59842b5674ebcb4d907921e584f5f2)) but i'm not sure musl is a good idea for security sensitive projects.

This will use a native rust tls implementation with a fixed version, if we would have regular builds linked against debian's openssl we would inherit security fixes, by sacrificing a few megabytes (17mb musl vs 40mb distroless)

Reply:
My intention is to have a container image without shell in it. Be it scratch or something else an image without shell is secure especially if it has socket exposed to it.

Reply:
Ah, but that should be addressed by distroless, no?

Reply:
Ah right, makes sense. Yeah distroless should work too.